### ISSUE TITLE
A Soundfont Request

### ISSUE THREAD

**Issue by PencilPeach** [yliaho/alundra-preservation-project#1]
Hello! Sorry to bother you, and I know this project hasn't been visited in years, but I figured I'd ask regardless

If it's not too much trouble, I was wondering if it would be possible for you to rip the soundfont for The Shrine of The Lake and Kline's Dream from the Alundra OST?
I recently played the game myself sometime last year and I really fell in love with it! I think your dedication to preserving it is really admirable, it's a very unique game!

Anyway, I hope you're doing alright regardless, and even if you don't or can't rip the soundfont, I appreciate the work you've done on this project regardless! I hope you're doing well!

**Comment by miyl** [comment]
It's not my repo, but I wanted to help also because I'm curious it myself, and I'm working on an Alundra remix just for fun.
Unfortunately I can't really help yet, except provide a bit of info, for you or anyone else reading along.

On the Alundra disc there's a DATAS.BIN, which practically contains all the data of the game, but unfortunately I haven't yet found a tool to extract the files from it.
This repo - https://github.com/surixurient/alundra - has some programs which may be able to do so, but it's not really documented.
...and this preservation repo doesn't document how to extract those files either. Its README was also written before surixurient resurfaced and made the above repo.

The following repo has a tool called `alundat` but I'm not sure exactly what it does. It takes DATAS.BIN and does some operations on it, possibly removing its obfuscation so other programs can extract the files. But its output is still a single file. I tried, subsequently, to extract the files with `binwalk` but that didn't do anything:
https://github.com/suppertails66/wdtools/

There IS one person here talking about having another extraction tool which is apparently written in Russian, which I don't understand, but maybe its simple enough so randomly clicking around is a viable strategy. Don't have that tool myself yet, so I don't know if that would provide the soundfonts. Also the user didn't share the program, so one would have to write to them and hope to get a response. I will do that:
https://forum.xentax.com/viewtopic.php?p=140930&sid=05b6b39d6744fd09c76de29c72f9da22#p140930

I'm curious what you want the soundfont for? Making some music?

**Comment by PencilPeach** [comment]
Thank you very much for the response and the information! Hopefully you get a response quickly!

And I'd like the soundfont for making music, yes! I'm learning how to compose music for a game I'm making and I really love the soundfonts of my favorite PSX games like Alundra and Breath of Fire 3/4, so I would really love to have the soundfonts to play around with to make songs for it! I've been trying my best to find a way to get them but PSX soundfonts aren't ripped/shared as much as SNES soundfonts X0

Thank you again for the info! I appreciate you taking time out of your day to respond! I hope you're having a good day!

**Comment by miyl** [comment]
Hi Pencil,

That sounds great! Good games!

I found out the music and sounds are apparently in SOUND.BIN instead, rather than DATAS.BIN, but as I can understand it the headers for the files are elsewhere, which might be DATAS.BIN. The two have to be joined to get the proper files.
The suxurient repo contains code for dealing with those sounds but it's written in C# which I'm not that familiar with, and there aren't instructions on compiling it. Not sure which version of .NET is needed fx.

Fortunately I found out that instead, one can find the soundtrack in the format called `PSF`, fx. on zophar.net
That format contains the songs in a `MIDI`-iike format, along with the sounds used in them.
With **VGMTrans** - which I found out was also used for the Alundra sound files in this repo - you can extract the MIDI, soundfont and wave files from the `PSF`:
https://github.com/vgmtrans/vgmtrans

Basically you just: 
1. Get **VGMTrans**
2. Get the Alundra `PSF` file(s)
3. Open that `PSF` file inside **VGMTrans**
4. You can right-click each of the three data types it identifies, and choose to export them to `MIDI`/Soundfont/`WAV`! 
5. Voila!

Thank you, I hope you're having a good day as well!
